mosn config dumper dump err when rename config file

### Describe the bug

When mosn run in container, `/tmp/` and `/home/admin/mosn/conf/` may use different file system, then rename may produce `invalid cross-device link` error.

### Expected behavior
`/tmp/write_safe_temp` rename to `/home/admin/mosn/conf/mosn_config.json` can work fine.

### Actual behavior
Log:
```
[ERROR] dump config failed, caused by: rename /tmp/write_safe_temp /home/admin/mosn/conf/mosn_config.json: invalid cross-device link
```

### Steps to reproduce

### Minimal yet complete reproducer code (or GitHub URL to code)

### Environment
+ SOFAMosn 0.4.1

--- a/pkg/config/dump_test.go
+++ b/pkg/config/dump_test.go
@@ -20,6 +20,7 @@ package config
 import (
 	"bytes"
 	"io/ioutil"
+	"os"
 	"testing"
 )
 
@@ -36,5 +37,37 @@ func TestWriteFileSafety(t *testing.T) {
 	}
 	if !bytes.Equal(data, b) {
 		t.Error("write data is not expected")
+	}
+}
+
+func TestCopyFile(t *testing.T) {
+	// when
+	src := "/tmp/test_write_file_safety_src"
+	data := []byte("test_data")
+	if err := ioutil.WriteFile(src, data, 0644); err != nil {
+		t.Fatal("write test src file error: ", err)
+	}
+
+	// then
+	target := "/tmp/test_write_file_safety_target"
+	if err := CopyFile(src, target); err != nil {
+		t.Fatal("copy file error: ", err)
+	}
+
+	// verify
+	b, err := ioutil.ReadFile(target)
+	if err != nil {
+		t.Fatal("read target file failed: ", err)
+	}
+	if !bytes.Equal(data, b) {
+		t.Error("write data is not expected")
+	}
+
+	f, err := os.Stat(target)
+	if err != nil {
+		t.Fatal("read target file stat failed: ", err)
+	}
+	if !(f.Mode() == 0644) {
+		t.Fatal("target file stat verify failed: ", f.Mode())
 	}
 }

--- a/pkg/config/dump_test.go
+++ b/pkg/config/dump_test.go
@@ -38,30 +38,6 @@ func TestWriteFileSafety(t *testing.T) {
 	if !bytes.Equal(data, b) {
 		t.Error("write data is not expected")
 	}
-}
-
-func TestCopyFile(t *testing.T) {
-	// when
-	src := "/tmp/test_write_file_safety_src"
-	data := []byte("test_data")
-	if err := ioutil.WriteFile(src, data, 0644); err != nil {
-		t.Fatal("write test src file error: ", err)
-	}
-
-	// then
-	target := "/tmp/test_write_file_safety_target"
-	if err := CopyFile(src, target); err != nil {
-		t.Fatal("copy file error: ", err)
-	}
-
-	// verify
-	b, err := ioutil.ReadFile(target)
-	if err != nil {
-		t.Fatal("read target file failed: ", err)
-	}
-	if !bytes.Equal(data, b) {
-		t.Error("write data is not expected")
-	}
 
 	f, err := os.Stat(target)
 	if err != nil {

--- a/pkg/config/dumper.go
+++ b/pkg/config/dumper.go
@@ -18,6 +18,7 @@
 package config
 
 import (
+	"io"
 	"io/ioutil"
 	"os"
 	"sync"
@@ -154,6 +155,46 @@ Try:
 	if err != nil {
 		return err
 	}
-	err = os.Rename(tempFile, filename)
+	err = CopyFile(tempFile, filename)
+	return
+}
+
+// credit https://gist.github.com/r0l1/92462b38df26839a3ca324697c8cba04
+func CopyFile(src, dst string) (err error) {
+	in, err := os.Open(src)
+	if err != nil {
+		return
+	}
+	defer in.Close()
+
+	out, err := os.Create(dst)
+	if err != nil {
+		return
+	}
+	defer func() {
+		if e := out.Close(); e != nil {
+			err = e
+		}
+	}()
+
+	_, err = io.Copy(out, in)
+	if err != nil {
+		return
+	}
+
+	err = out.Sync()
+	if err != nil {
+		return
+	}
+
+	si, err := os.Stat(src)
+	if err != nil {
+		return
+	}
+	err = os.Chmod(dst, si.Mode())
+	if err != nil {
+		return
+	}
+
 	return
 }

--- a/pkg/config/dumper.go
+++ b/pkg/config/dumper.go
@@ -18,7 +18,6 @@
 package config
 
 import (
-	"io"
 	"io/ioutil"
 	"os"
 	"sync"
@@ -141,10 +140,9 @@ func DumpConfigHandler() {
 	})
 }
 
-const tempFile = "/tmp/write_safe_temp"
-
 // WriteFileSafety trys to over write a file safety.
 func WriteFileSafety(filename string, data []byte, perm os.FileMode) (err error) {
+	tempFile := filename + ".tmp"
 Try:
 	for i := 0; i < 5; i++ {
 		err = ioutil.WriteFile(tempFile, data, perm)
@@ -155,46 +153,6 @@ Try:
 	if err != nil {
 		return err
 	}
-	err = CopyFile(tempFile, filename)
-	return
-}
-
-// credit https://gist.github.com/r0l1/92462b38df26839a3ca324697c8cba04
-func CopyFile(src, dst string) (err error) {
-	in, err := os.Open(src)
-	if err != nil {
-		return
-	}
-	defer in.Close()
-
-	out, err := os.Create(dst)
-	if err != nil {
-		return
-	}
-	defer func() {
-		if e := out.Close(); e != nil {
-			err = e
-		}
-	}()
-
-	_, err = io.Copy(out, in)
-	if err != nil {
-		return
-	}
-
-	err = out.Sync()
-	if err != nil {
-		return
-	}
-
-	si, err := os.Stat(src)
-	if err != nil {
-		return
-	}
-	err = os.Chmod(dst, si.Mode())
-	if err != nil {
-		return
-	}
-
+	err = os.Rename(tempFile, filename)
 	return
 }

